data/data_test.go

```diff
@@ -0,0 +1,132 @@
+package data
+
+import (
+	"errors"
+	"reflect"
+	"testing"
+)
+
+type TestClient struct {
+	*MeetupClient
+
+	data []byte
+	err  error
+}
+
+func NewClient(data string, err error) TestClient {
+	return TestClient{
+		data: []byte(data),
+		err:  err,
+	}
+}
+
+func (c TestClient) Get(key string) ([]byte, error) {
+	if c.err != nil {
+		return []byte{}, c.err
+	}
+
+	return c.data, c.err
+}
+
+func TestScheduleHasNoEvents(t *testing.T) {
+	s := Schedule{}
+	if len(s.Events) != 0 {
+		t.Fail()
+	}
+}
+
+func TestFetchEventsStoresEvents(t *testing.T) {
+	s := Schedule{key: "Boulder-Gophers"}
+
+	c := NewClient(`[{"id":"id","name":"Event","time":400}]`, nil)
+
+	s.FetchEvents(c)
+
+	expected := []Event{Event{ID: "id", Name: "Event", Time: 400}}
+
+	if !reflect.DeepEqual(s.Events, expected) {
+		t.Fail()
+	}
+}
+
+func TestFetchEventsReturnsErrorWhenRequestFails(t *testing.T) {
+	s := Schedule{key: "Boulder-Gophers"}
+	c := NewClient(``, errors.New("Failed to connect"))
+
+	err := s.FetchEvents(c)
+
+	if err == nil {
+		t.Fail()
+	}
+}
+
+func TestFetchEventsReturnsErrorWhenUnmarshalFails(t *testing.T) {
+	s := Schedule{key: "Boulder-Gophers"}
+	c := NewClient(`<>`, nil)
+
+	err := s.FetchEvents(c)
+
+	if err == nil {
+		t.Fail()
+	}
+}
+
+func TestFetchEventsReturnsEventsInOrderOfTime(t *testing.T) {
+	s := Schedule{key: "Boulder-Gophers"}
+	c := NewClient(`
+		[
+			{"id":"two","name":"Two","time":2},
+			{"id":"one","name":"One","time":1}
+		]
+	`, nil)
+
+	s.FetchEvents(c)
```

Comment:
You should assert that the error returned her is nil.

Reply:
👍